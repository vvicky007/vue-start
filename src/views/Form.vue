<template>
  <div class="panel">
    <form action="" novalidate class="form">
      <div>
        <label for="email">Email</label>
        <input
          v-validate="{ val: this.email, type: 'email' }"
          type="email"
          name="email"
          id="email"
          @change="updateEmail($event.target.value)"
          :value="email"
        />
      </div>
      <div>
        <label for="pwd">Password</label>
        <input
          type="password"
          name="pwd"
          id="pwd"
          @change="updatePassword($event.target.value)"
          :value="password"
        />
      </div>
      <button>Submit</button>
    </form>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import Validate from "@/shared/validate";
@Component({
  name: "Form",
  components: {},
  directives: {
    validate: Validate,
  },
})
export default class Form extends Vue {
  email = "";
  password = "";
  updateEmail(email: string): void {
    this.email = email;
  }
  updatePassword(password: string): void {
    this.password = password;
  }
}
</script>
<style lang="scss" scoped>
.panel {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  justify-content: center;
  height: 100%;
}
.form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  justify-items: center;
}
button {
  background-color: #42b983;
  width: 40%;
  margin-top: 10px;
  margin-left: auto;
  margin-right: auto;
}
.form > div {
  margin: 10px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: center;
}
.form > div > label {
  margin-top: 5px;
  margin-right: 5px;
}
</style>
