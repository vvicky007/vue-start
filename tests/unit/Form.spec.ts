import { shallowMount, mount, createLocalVue, Wrapper } from "@vue/test-utils";
import Form from "@/views/Form.vue";
describe("Form.vue", () => {
  let wrapper: Wrapper<Form>;
  beforeEach(() => {
    wrapper = mount(Form);
  });
  it("Should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Should contain an input", () => {
    const inp = wrapper.findAll("input");
    expect(inp.length).toBe(2);
  });
  it("should be red when an invalid input is given", async () => {
    const inp = wrapper.findAll("input");
    await inp.at(0).setValue("dawd");
    await inp.at(0).trigger("change");
    expect(inp.at(0).attributes().style).toBe("border: 2px solid red;");
  });
  it("should be normal when an valid input is given", async () => {
    const inp = wrapper.findAll("input");
    await inp.at(0).setValue("dawd@");
    await inp.at(0).trigger("change");
    expect(inp.at(0).attributes().style).toBe("");
  });
  it("should take user password", async () => {
    const inp = wrapper.findAll("input");
    await inp.at(1).setValue("dawd@");
    await inp.at(1).trigger("change");
    expect(wrapper.vm.$data.password).toBe("dawd@");
  });
  afterEach(() => {
    wrapper.destroy();
  });
});
