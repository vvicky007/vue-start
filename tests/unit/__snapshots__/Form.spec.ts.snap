// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.vue Should match the snapshot 1`] = `
<div class="panel">
  <form action="" novalidate="novalidate" class="form">
    <div><label for="email">Email</label> <input type="email" name="email" id="email" style="border: 2px solid red;"></div>
    <div><label for="pwd">Password</label> <input type="password" name="pwd" id="pwd"></div> <button>Submit</button>
  </form>
</div>
`;
